spec/unit/provider/package/zypper_spec.rb

```diff
@@ -163,6 +165,15 @@ def shell_out_expectation!(*command, **options)
       )
       provider.install_package(["emacs"], ["1.0"])
     end
+
+    it "should run zypper install with source option" do
+      new_resource.source "/tmp/wget_1.11.4-1ubuntu1_amd64.rpm"
+      allow(::File).to receive(:exist?).with("/tmp/wget_1.11.4-1ubuntu1_amd64.rpm").and_return(true)
+      shell_out_expectation!(
+        "zypper", "--non-interactive", "install", "--auto-agree-with-licenses", "--oldpackage", "/tmp/wget_1.11.4-1ubuntu1_amd64.rpm"
+      )
+      provider.install_package(["emacs"], ["1.0"])
```

Comment:
these tests should have the correct behavior from the superclass which i believe is:

```      
provider.install_package(["wget"], ["1.11.4-1ubuntu1_amd64"])
```

or something close to that.

this is also insufficient for testing the actual cooperation between the superclass and the subclass and making sure that the subclass is filling out the `candidate_version` and `installed_version` correctly along with the correct `package_name_array` so that the call from the superclass into `install_package` has the correct information.

that needs to be done correctly, even if the install_package routine goes and trusts the source, because the idempotent behavior is handled by the superclass and that is what needs to be correct.